appinventor/components/src/com/google/appinventor/components/runtime/util/ListViewArrayAdapterTwoTextLinear.java

```diff
@@ -0,0 +1,138 @@
+
+/**
+ * Adapter class to populate ListView with two lines of text in a single main line
+ */
+public class ListViewArrayAdapterTwoTextLinear {
+
+  private int textSize, detailTextSize, textColor, detailTextColor;
+  private static ComponentContainer container;
+  private ArrayList<JSONObject> currentItems;
+  private ArrayList<JSONObject> filterCurrentItems;
+
+  public ListViewArrayAdapterTwoTextLinear(int textSize, int detailTextSize, int textColor, int detailTextColor,
+      ComponentContainer container, ArrayList<JSONObject> currentItems) {
+    this.textSize = textSize;
+    this.detailTextSize = detailTextSize;
+    this.textColor = textColor;
+    this.detailTextColor = detailTextColor;
+    this.container = container;
+    this.currentItems = new ArrayList<>(currentItems);
+    this.filterCurrentItems = new ArrayList<>(currentItems);
+  }
+
+  /**
+   * method to create view for each row of ListView
+   *
+   * @return parent container containing two textViews
+   */
+  private View createView(){
+    LinearLayout linearLayout = new LinearLayout(container.$context());
+    linearLayout.setId(1);
+    linearLayout.setOrientation(LinearLayout.HORIZONTAL);
+    linearLayout.setPadding(15,15,15,15);
+    TextView textView1 = new TextView(container.$context());
+    textView1.setPadding(10,10,10, 10);
+    textView1.setGravity(Gravity.LEFT);
+    textView1.setLayoutParams(new LinearLayout.LayoutParams(450, ViewGroup.LayoutParams.WRAP_CONTENT));
+    textView1.setId(2);
+    TextView textView2 = new TextView(container.$context());
+    textView2.setPadding(10,10,10, 10);
+    textView2.setGravity(Gravity.RIGHT);
+    textView2.setLayoutParams(new LinearLayout.LayoutParams(450, ViewGroup.LayoutParams.WRAP_CONTENT));
+    textView2.setId(3);
+    linearLayout.addView(textView1);
+    linearLayout.addView(textView2);
+    return linearLayout;
+  }
+
+  public ArrayAdapter<JSONObject> createAdapter() {
+    ArrayAdapter<JSONObject> itemAdapter = new ArrayAdapter<JSONObject>(container.$context(), 0, currentItems) {
+      @Override
+      public View getView(int position, View convertView, ViewGroup parent) {
+        View view = createView();
+
+        TextView text1 = view.findViewById(2);
+        TextView text2 = view.findViewById(3);
+
+        JSONObject row = filterCurrentItems.get(position);
+        String val1 = row.has("Text1") ? row.getString("Text1") : "";
+        String val2 = row.has("Text2") ? row.getString("Text2") : "";
+
+        text1.setText(val1);
+        text2.setText(val2);
+
+        text1.setTextColor(textColor);
+        text2.setTextColor(detailTextColor);
+
+        text1.setTextSize(textSize);
+        text2.setTextSize(detailTextSize);
+
+        return view;
+      }
+
+      /*
+       * overriding getFilter() method to implement search functionality for the custom layout of ListView
+       */
+      @Override
+      public Filter getFilter() {
+        Filter filter = new Filter() {
```

Comment:
A call to getFilter will construct a new filter object every time. For memory reasons, it might make sense to move this to a final variable on the class and then return that object (so it is only constructed once per ListView object.